### PR TITLE
Fix test failures in release 1.1 branch

### PR DESCRIPTION
**Pull request by Priya91**
The System.Net.Requests tests are segfaulting in release branches as the latest System.Net.Http lib is not being used for tests. Since the [fix](https://github.com/dotnet/corefx/commit/63bc3dbff7754dc225efdb1c392ec0af8c405a5a) in System.Net.Http is not yet shipped, the stable package which the test uses is incorrect. Hence, add a direct project reference to System.Net.Http from System.Net.Requests.Tests.

Also bump the package and assembly version of System.Net.Http, to put it in the release path.

cc @weshaggard @danmosemsft

### CODE REVIEW

**Comment by danmosemsft** [comment]
Seems fine but I'm not sufficient to review this...

Did you devise any way to find whether there could be other cases like this? Perhaps we are simply relying on tests to find them, just as they "found" this one.

**Comment by Priya91** [comment]
> Did you devise any way to find whether there could be other cases like this?

I went through the history of the release/1.1 branch, this [commit](https://github.com/dotnet/corefx/commit/9e98f24f508d2cda97f10474a86052f51bc9c7e3), which is an update after the servicing release. From then these are the concrete product commits,

1. [Interop.MountPoints.cs](https://github.com/dotnet/corefx/commit/3d2debcf3f4fb7ce06f58c0b617348835e9036c8) by @wfurt 
2. [WinHttpHandler](https://github.com/dotnet/corefx/commit/454bdeb2ef477344b8626307892f442bff2a3106) by @davidsh 
3. [Crypto native shim](https://github.com/dotnet/corefx/commit/8e1d507481334132d17aad913b3f8aebdefb5c9e) by @bartonjs , versioning fixed with this [commit](https://github.com/dotnet/corefx/commit/a47f66b09ade75b0cecf76c02c13754e93935326) and [this](https://github.com/dotnet/corefx/commit/2ef02bda627168fd307ac229152234745ca56f4a)
4. [System.Net.Http](https://github.com/dotnet/corefx/commit/63bc3dbff7754dc225efdb1c392ec0af8c405a5a) by @bartonjs - versioning addressed in this PR.

The first 2 versions are not updated.

**Comment by danmosemsft** [comment]
Would you mind getting those 2 fixed also then? I'm guessing same in 1.0

**Comment by davidsh** [comment]
>We should think about whether or not we also need to ship and update to the System.Net.Requests (or any other package that depends on System.Net.Http) that require this update.

Yes, you should make sure that System.Net.Requests will depend on the updated System.Net.Http package.

**Comment by Priya91** [comment]
@weshaggard System.Net.Requests does require this update, I dunno how to make System.Net.Requests depend on this version of System.Net.Http, should i make the project.json of System.Net.Requests reference 4.3.2 package, can you link an example?

**Comment by weshaggard** [comment]
>@weshaggard System.Net.Requests does require this update, I dunno how to make System.Net.Requests depend on this version of System.Net.Http, should i make the project.json of System.Net.Requests reference 4.3.2 package, can you link an example?

I don't have a pointer to an example but I think the easiest way to get this is to take a ProjectReference to System.Net.Http and start building the package for System.Net.Requests as well as bumping its Assembly and Package version.

**Comment by Priya91** [comment]
@weshaggard Can you please checkout the latest commit for the System.Net.Request change.

File: src/System.Net.Requests/dir.props [hunk 1]
```diff
@@ -1,7 +1,8 @@
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="..\dir.props" />
   <PropertyGroup>
-    <AssemblyVersion>4.0.12.0</AssemblyVersion>
+    <AssemblyVersion>4.0.13.0</AssemblyVersion>
```
Comment (weshaggard) [inline src/System.Net.Requests/dir.props]: For servicing we generally bump the 4th part of the assembly version.

File: src/System.Net.Requests/src/System.Net.Requests.csproj [hunk 2]
```diff
@@ -49,6 +49,10 @@
     <Compile Include="$(CommonPath)\System\NotImplemented.cs">
       <Link>Common\System\NotImplemented.cs</Link>
     </Compile>
+    <ProjectReference Include="..\..\System.Net.Http\pkg\System.Net.Http.pkgproj">
```
Comment (weshaggard) [inline src/System.Net.Requests/src/System.Net.Requests.csproj]: I believe this is correct but I cannot recall for sure. I remember at some points we needed to reference the csproj and in some cases we reference the pkgproj. If the CI passes with the pkgproj we can keep it like it is.

File: src/System.Net.Requests/dir.props (hunk 1 above)
Comment (Priya91) [inline src/System.Net.Requests/dir.props]: so it should be 4.0.12.1 ?

Comment (weshaggard) [inline src/System.Net.Requests/dir.props]: Yes

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 3]
```diff
@@ -1479,7 +1480,8 @@
         "3.9.0.0": "3.9.0",
         "4.0.0.0": "4.0.0",
         "4.0.10.0": "4.0.10",
-        "4.0.11.0": "4.0.11"
+        "4.0.11.0": "4.0.11",
+        "4.0.12.0": "4.3.0"
```
Comment (Priya91) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: @weshaggard The build throws error when adding 4.0.12.1: 4.3.1, asks to run /t:updatepackageindex target. Upon running that target, this is the change generated.

**Comment by Priya91** [comment]
> Would you mind getting those 2 fixed also then? I'm guessing same in 1.0

@danmosemsft I've filed issue #23119 to track those changes.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json (hunk 3 above)
Comment (danmosemsft) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: @ericstj can you okay this in @weshaggard absence? Would like to merge this today.

**Comment by Priya91** [comment]
@danmosemsft I'm fairly confident these changes are correct, the builds also passed. On my local box, it has generated System.Net.Http.4.3.3-servicing-25610.nupkg, and System.Net.Requests.4.3.1-servicing-25610.nupkg for package build.

**Comment by danmosemsft** [comment]
Sounds good. Can you send mail to shiproom for this and the 1.0 one? I'll approve them, but I want changes into 1.x to get visibility there.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json (hunk 3 above)
Comment (weshaggard) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: The changes look correct. HOwever I would also expect to see a "4.0.12.1" to "4.3.1" mapping. However if not one is actually depending on the latest version it doesn't get used by anything.

File: src/packages.builds [hunk 4]
```diff
@@ -40,6 +40,12 @@
       <AdditionalProperties>$(AdditionalProperties);SkipCreatePackageOnMissingFiles=true</AdditionalProperties>
       <BuildAllOSGroups>$(BuildAllOSGroups)</BuildAllOSGroups>
     </Project>
+    <Project Include="System.Net.Http\pkg\System.Net.Http.builds">
+      <AdditionalProperties>$(AdditionalProperties)</AdditionalProperties>
+    </Project>
+    <Project Include="System.Net.Requests\pkg\System.Net.Requests.builds">
+      <AdditionalProperties>$(AdditionalProperties)</AdditionalProperties>
+    </Project>
```
Comment (bartonjs) [inline src/packages.builds]: These two got put in the native packages section instead of the managed packages section...

**Pull request merged**